pii_recognition/evaluation/character_level_evaluation_test.py

```diff
@@ -0,0 +1,230 @@
+from typing import List
+
+import pytest
+from pii_recognition.evaluation.character_level_evaluation import (
+    compute_entity_precisions_for_prediction,
+    compute_entity_recalls_for_ground_truth,
+    label_encoder,
+)
+from pii_recognition.labels.schema import Entity
+
+
+def test_label_encoder_for_0_taken():
+    with pytest.raises(ValueError) as err:
+        label_encoder(3, [], {"LOC": 0})
+    assert str(err.value) == (
+        "Value 0 is reserved! If a character does not "
+        "belong to any entity, it would be assigned with 0."
+    )
+
+
+def test_label_encoder_for_multi_labels():
+    spans = [
+        Entity(entity_type="LOC", start=5, end=8),
+        Entity(entity_type="PER", start=10, end=15),
+        Entity(entity_type="PERSON", start=2, end=5),
+    ]
+
+    # entity PER and PERSON map to the same int
+    actual = label_encoder(20, spans, {"LOC": 1, "PER": 2, "PERSON": 2})
+    assert actual == [0, 0, 2, 2, 2, 1, 1, 1, 0, 0, 2, 2, 2, 2, 2, 0, 0, 0, 0, 0]
+
+
+def test_label_encoder_for_missing_label_in_mapping():
+    spans = [
+        Entity(entity_type="LOC", start=5, end=8),
+        Entity(entity_type="PER", start=10, end=15),
+    ]
+
+    with pytest.raises(Exception) as error:
+        label_encoder(20, spans, {"LOC": 1})
+    assert str(error.value) == (
+        "Missing label 'PER' in 'label_to_int' mapping."
+    )
+
+
+def test_label_encoder_for_span_beyond_range():
+    spans = [Entity(entity_type="LOC", start=3, end=7)]
+
+    with pytest.raises(ValueError) as error:
+        label_encoder(5, spans, {"LOC": 1})
+    assert str(error.value) == (
+        "Span index is out of range: text length is 5 but got span index 7."
+    )
+
+
+def test_compute_precisions_recalls_for_exact_match():
```

Comment:
I think you should expand this test to include more than one entity and more than one entity type. This will better test the loop logic as well as make sure it works in a multiclass scope

Reply:
Done. All test cases updated regarding to precision and recall calculations.